src/instructlab/rag/haystack/component_factory.py

```diff
@@ -0,0 +1,61 @@
+# Third Party
+from haystack.components.converters import TextFileToDocument  # type: ignore
+from haystack.components.embedders import (  # type: ignore
+    SentenceTransformersTextEmbedder,
+)
+from milvus_haystack import MilvusDocumentStore  # type: ignore
+from milvus_haystack.milvus_embedding_retriever import (  # type: ignore
+    MilvusEmbeddingRetriever,
+)
+
+# First Party
+from instructlab.rag.haystack.components.document_splitter import (
+    DoclingDocumentSplitter,
+)
+from instructlab.rag.rag_configuration import (
+    DocumentStoreConfig,
+    EmbeddingModelConfig,
+    RetrieverConfig,
+)
+
+
+def create_document_store(document_store_config: DocumentStoreConfig, drop_old: bool):
```

Comment:
passing these types of instances is generally fine unless we intend to call these types of functions in subprocess eventually (some sort of ilab chat --rag -dt). I trust your judgement on when the entire obj is needed and when only certain parts are needed. Generally, we are trying to make our args light weight for future-proofing.